packages/jsii-rosetta/test/commands/extract.test.ts

```diff
@@ -59,6 +59,45 @@ beforeEach(() => {
 
 afterEach(() => assembly.cleanup());
 
+test('extract works with zipped assembly files', async () => {
+  const assembly = TestJsiiModule.fromSource(
+    {
+      'index.ts': `
+      export class ClassA {
+        public someMethod() {
+        }
+      }
+      export class ClassB {
+        public anotherMethod() {
+        }
+      }
+      `,
+      'README.md': DUMMY_README,
+    },
+    {
+      name: 'my_assembly',
+      jsii: DUMMY_JSII_CONFIG,
+    },
+    true /* zipAssemblyFile */,
+  );
+
+  // assert that assembly is zipped
+  expect(fs.existsSync(path.join(assembly.moduleDirectory, '.jsii.gz'))).toBeTruthy();
+  expect(fs.existsSync(path.join(assembly.moduleDirectory, '.jsii'))).toBeFalsy();
+
+  // behavior is as expected
+  const cacheToFile = path.join(assembly.moduleDirectory, 'test.tabl.json');
```

Comment:
FYI this file will be HUGE, and is worth compressing, too.

Reply:
definitely will be compressing the tablet file too, but in a separate PR :)